### PR TITLE
ci: combine workflows and homogenize testing target names; limit concurrency

### LINKED ISSUE #13: Improve the GitHub Actions to reduce significant duplication in workflow files
Pull requests https://github.com/google/globalfoundries-pdk-libs-gf180mcu_fd_pr/pull/12 and https://github.com/google/globalfoundries-pdk-libs-gf180mcu_fd_pr/pull/11 added a whole bunch of GitHub Actions for checking various things.

The workflow files currently have large amount of duplicate in them. Instead we should be using "matrix" for this stuff. See the documentation at https://github.com/google/globalfoundries-pdk-libs-gf180mcu_fd_pr/pull/11

### PR DESCRIPTION
**Pull request by umarcor**
Fixes #13

/cc @mithro

### CODE REVIEW

**Review by mohanad0mohamed — APPROVED** [review]
@umarcor Great job!

- The `makefile` now looks cleaner. 
- The `Tests.yml` is doing the same jobs as the separate files was.
- I'm not sure if removing comments was a good choice. it would be easier for other developers to understand what is written if they were there.

**Review by atorkmabrains — APPROVED** [review]
@umarcor and @mithro Changes looks great from my end. Could you please merge?

**Comment by atorkmabrains** [comment]
@mohanad0mohamed Could we create a new PR to split the main DRC to smaller parts as well?

**Comment by mohanad0mohamed** [comment]
@atorkmabrains There is already an issue about that request: https://github.com/google/globalfoundries-pdk-libs-gf180mcu_fd_pr/issues/15

**Review by atorkmabrains — APPROVED** [review]
All changes seem great. And I approve changes.

File: .github/workflows/Tests.yml [hunk 1]
```diff
@@ -14,6 +14,10 @@
 
 name: Regression testing
 
+concurrency:
```
Comment (mithro) [inline .github/workflows/Tests.yml:19]: Could you add a comment about why we want to limit concurrency?

File: rules/klayout/drc/testing/Makefile [hunk 2]
```diff
@@ -23,16 +23,16 @@ Testing_DIR ?= $(shell pwd)
 
 all : clean test-DRC
 
-test-DRC: test-DRC_switch  test-DRC_SC  test-DRC_main
+test-DRC: test-DRC-switch  test-DRC-SC  test-DRC-main
```
Comment (mithro) [inline rules/klayout/drc/testing/Makefile:26]: The changes to these `Makefile` should probably be a separate commit?

**Review by mithro — APPROVED** [review]
Two very minor comments.

File: rules/klayout/drc/testing/Makefile (hunk 2 above)
Comment (atorkmabrains) [reply rules/klayout/drc/testing/Makefile:26]: @mithro these changes are done by @umarcor to make it cleaner for github actions. Doesn't affect the main functionality from my end.

File: .github/workflows/Tests.yml (hunk 1 above)
Comment (atorkmabrains) [reply .github/workflows/Tests.yml:19]: @umarcor added this. Not sure why. Could you please approve this?

File: rules/klayout/drc/testing/Makefile (hunk 2 above)
Comment (umarcor) [reply rules/klayout/drc/testing/Makefile:26]: See https://github.com/google/globalfoundries-pdk-libs-gf180mcu_fd_pr/pull/28.

File: .github/workflows/Tests.yml (hunk 1 above)
Comment (umarcor) [reply .github/workflows/Tests.yml:19]: Added a reference to the docs about the concurrency feature  in GitHub Actions: https://github.com/google/globalfoundries-pdk-libs-gf180mcu_fd_pr/pull/18/commits/8d7f2e455e8cf10fcb71ac6501ed5e72a2a04916.

Comment (umarcor) [reply .github/workflows/Tests.yml:19]: Done in https://github.com/google/globalfoundries-pdk-libs-gf180mcu_fd_pr/pull/18/commits/8d7f2e455e8cf10fcb71ac6501ed5e72a2a04916.

**Review by atorkmabrains — COMMENTED** [review]
@umarcor You moved the klayout version to klayout 0.27.10 we want to use klayout 0.27.8.

**Comment by umarcor** [comment]
@atorkmabrains I removed the commit bumping klayout.

File: .github/workflows/Tests.yml [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: Regression testing
+
+# https://docs.github.com/en/actions/using-jobs/using-concurrency
```
Comment (mithro) [inline .github/workflows/Tests.yml:18]: Rather than linking to the concurrency documentation, can you explain why we want / need the concurrency statement?

File: rules/klayout/drc/testing/Makefile (hunk 2 above)
Comment (mithro) [reply rules/klayout/drc/testing/Makefile:26]: #28 looks good. Will merge if CI is green.

**Review by mithro — APPROVED** [review]
LGTM! Will merge if CI goes green.

File: .github/workflows/Tests.yml (hunk 3 above)
Comment (umarcor) [reply .github/workflows/Tests.yml:18]: Done in https://github.com/google/globalfoundries-pdk-libs-gf180mcu_fd_pr/pull/18/commits/14838d309728f06c81f4af8718eb55202d320987.

**Review by mithro — APPROVED** [review]
